library/core/src/task/wake.rs

```diff
@@ -782,12 +796,28 @@ impl LocalWaker {
         Self { waker }
     }
 
-    /// Creates a new `LocalWaker` that does nothing when `wake` is called.
+    /// Returns a reference to a `LocalWaker` that does nothing when used.
     ///
+    // Note!  Much of the documentation for this method is duplicated
+    // in the docs for `Waker::noop`.
+    // If you edit it, consider editing the other copy too.
+    //
     /// This is mostly useful for writing tests that need a [`Context`] to poll
     /// some futures, but are not expecting those futures to wake the waker or
     /// do not need to do anything specific if it happens.
     ///
+    /// If an owned `LocalWaker` is needed, `clone()` this one.
+    ///
+    /// # Warning
+    ///
+    /// Do not use this function as a way to be able to call `poll` methods,
+    /// in production code,
+    /// from within a non-`async`, non-`poll`, function.
+    ///
+    /// Using a no-op waker for that purpose would cause wakeups to be lost:
```

Comment:
"cause wakeups to be lost" feels very imprecise? Who did the wakeups belong to that they are now lost from? "The proper waker", sure, but it feels like it just obscures the meaning of the rest of this sentence.